README_テーブルの情報

kenshiro-wata opened the pull request:
ER図こちらになります。よろしくお願いします！
https://drive.google.com/file/d/1Rs6wYbtBG3UfWdYvgYrt7mLvck5v-a6d/view?usp=sharing

kenshiro-wata committed:
README_テーブルの情報

syuji-yamamoto commented on README.md:
```diff
@@ -22,3 +22,61 @@ Things you may want to cover:
+| password                  | string | null: false |
+| confirmation password     | string | null: false |
+| last name                 | string | null: false |
+| first name                | string | null: false |
```
カラム名にスペースは好ましくありません。
命名規則に従ってアンダーバーにしましょう！

syuji-yamamoto replied:
姓名のフリガナを保存するカラムを用意しましょう！

syuji-yamamoto commented on README.md:
```diff
@@ -22,3 +22,61 @@ Things you may want to cover:
+| confirmation password     | string | null: false |
+| last name                 | string | null: false |
+| first name                | string | null: false |
+| birth                     | string | null: false |
```
誕生日カラムはdate型にしましょう！

syuji-yamamoto commented on README.md:
```diff
@@ -22,3 +22,61 @@ Things you may want to cover:
+| ------------------- | ---------- | ----------- |
+| image               |            | null: false |
+| name                | string     | null: false |
+| description         | string     | null: false |
```
商品説明を保存するカラムは、説明が長くなることを仮定すると、text型で設計した方が無難かと考えます。

syuji-yamamoto commented on README.md:
```diff
@@ -22,3 +22,61 @@ Things you may want to cover:
+| image               |            | null: false |
+| name                | string     | null: false |
+| description         | string     | null: false |
+| category            | string     | null: false |
+| item_status         | string     | null: false |
+| shipping_charges    | string     | null: false |
+| ship_from_address   | string     | null: false |
+| date_takes          | string     | null: false |
```
active_hashで実装しますので、integer型にしましょう！

syuji-yamamoto commented on README.md:
```diff
@@ -22,3 +22,61 @@ Things you may want to cover:
+| name                | string     | null: false |
+| description         | string     | null: false |
+| category            | string     | null: false |
+| item_status         | string     | null: false |
```
テーブル名で判断できるので、カラム名のitem_は不要です！

syuji-yamamoto commented on README.md:
```diff
@@ -22,3 +22,61 @@ Things you may want to cover:
+| ---------------------- | -----------| ----------- |
+| image                  |            | null: false |
+| name                   | string     | null: false |
+| credit_security_number | Int        | null: flase |
+| credit_valid           | Int        | null: flase |
+| credit_code            | Int        | null: flase |
```
セキュリティの観点から、クレジットカードの情報はデータベースに直接保存しないようにしましょう。

また、今回のフリマアプリではカードを登録する際は購入の度に入力となり、データベースにわざわざ記録しなくても良い情報となります。
ですので、これらのカラムを用意しなくても大丈夫です！

syuji-yamamoto commented on README.md:
```diff
@@ -22,3 +22,61 @@ Things you may want to cover:
+| credit_code            | Int        | null: flase |
+| address_city           | String     | null: flase |
+| block_number           | String     | null: flase |
+| phone_number           | Int        | null: flase |
```
電話番号は数字の文字列として実装しておきたいためstring型で実装しましょう！

kenshiro-wata replied:
なぜ数字としてではなく数字の文字列として扱うのでしょうか？

syuji-yamamoto replied:
頭に「０」がつく場合があるからです！
integer型だと頭の０は認識しないためです！

kenshiro-wata replied:
ありがとうございます！！

syuji-yamamoto commented on README.md:
```diff
@@ -22,3 +22,61 @@ Things you may want to cover:
+
+### Association
+
+- has_one :purchases
```
has_oneは単数系です！

syuji-yamamoto reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認ください！

kenshiro-wata committed:
テーブル設計（改善１）

kenshiro-wata committed:
README_テーブルの情報（改善１）

syuji-yamamoto commented on README.md:
```diff
@@ -23,32 +25,29 @@
 | ---------------------- | -----------| ----------- |
-| image                  |            | null: false |
+| image                  | string     | null: false |
 | name                   | string     | null: false |
```
配送先テーブルに名前と画像の情報は不要です！

syuji-yamamoto commented on README.md:
```diff
@@ -1,24 +1,56 @@
-# README
+# テーブル設計
```
商品は購入した後取引を行います、その際使用するのはuser_idとitem_idです！
それを管理するテーブルが必要ですので、設計してみましょう！

syuji-yamamoto reviewed (commented):
レビュー依頼ありがとうございます！
修正点を確認させていただきました！
コメントしましたので、ご確認ください！

kenshiro-wata committed:
README_テーブルの情報(改善２）

tenpadrummer commented on README.md:
```diff
@@ -1,24 +1,66 @@
 
-* System dependencies
+- has_many :items
+- has_many :deal
```
has_many :dealsですね！

tenpadrummer commented on README.md:
```diff
@@ -1,24 +1,66 @@
-* Database initialization
+| Column              | Type       | Options     |
+| ------------------- | ---------- | ----------- |
+| image               |            | null: false |
```
typeにstringを入れておきましょう！

tenpadrummer commented on README.md:
```diff
@@ -1,24 +1,66 @@
+| shipping_charges    | integer    | null: false |
+| ship_from_address   | integer    | null: false |
+| date_takes          | integer    | null: false |
+| price               | Int        | null: false |
```
ここだけIntになっているので、integerとしておきましょう！

tenpadrummer commented on README.md:
```diff
@@ -1,24 +1,66 @@
+| address_city           | String     | null: flase |
+| block_number           | String     | null: flase |
+| phone_number           | String     | null: flase |
+| item                   | references | null: false, foreign_key: true |
```
こちら、itemではなくdealではないでしょうか？
今回の設計では「配送先住所と購入管理が紐づく」実装を行なっていきたいので、商品やユーザーと配送先住所のアソシエーションは組まなくてもよいかと考えています。

tenpadrummer commented on README.md:
```diff
@@ -1,24 +1,66 @@
+- belongs_to :user
 
-* Deployment instructions
+## purchases テーブル
```
商品購入時に配送先住所を入力するさいに必要な市町村、番地、建物番号、郵便番号、電話番号のカラムが全て用意されているか確認しましょう！

tenpadrummer reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

kenshiro-wata committed:
README_テーブルの情報(改善3）

deal テーブルの名前をmanageに変更しました。
今一度ER図のチェックもお願いいたします。
https://drive.google.com/file/d/1Rs6wYbtBG3UfWdYvgYrt7mLvck5v-a6d/view?usp=sharing

syuji-yamamoto commented on README.md:
```diff
@@ -17,50 +17,52 @@
+- has_one :purchase
 
-## deal テーブル
+## purchases テーブル
```
都道府県を保存するカラムをinteger型で用意しましょう！

syuji-yamamoto commented on README.md:
```diff
@@ -17,50 +17,52 @@
 - belongs_to :user
 
-## purchases テーブル
+## manage テーブル
```
テーブル名は複数形にしましょう！

syuji-yamamoto commented on README.md:
```diff
@@ -1,24 +1,68 @@
+| name                      | string | null: false |
+| email                     | string | null: false |
+| password                  | string | null: false |
+| confirmation password     | string | null: false |
```
カラム名のスペースはアンダーバーに変更しておきましょう！

syuji-yamamoto reviewed (commented):
レビュー依頼ありがとうございます！
修正点を確認させていただきました！
コメントしましたので、ご確認ください！

kenshiro-wata committed:
README_テーブルの情報(改善3）

norito0627 commented:
レビュー依頼ありがとうございます！
LGTMとさせていただきます！実装お疲れ様でした！引き続き実装頑張っていきましょう！

kenshiro-wata commented:
ありがとうございました！！！
